File: 4Weak/Q5/Commando.py
from Soldier import Soldier
class Commando(Soldier):
    def __init__(self, name, baseAllowance, vocationAllowance):
        Soldier.__init__(self, name, baseAllowance)
        self.vocationAllowance = vocationAllowance

    def getGrossAllowance(self):
        return self.baseAllowance + self.vocationAllowance

    def __str__(self):
        return f"Commando{{{super().__str__()}, vocationAllowance={self.vocationAllowance}}}"

File: 4Weak/Q1/Commando.java
public class Commando extends Soldier {
    private double vocationAllowance;
    public Commando(String name, double baseAllowance, double vocationAllowance) {
        super(name, baseAllowance);
        this.vocationAllowance = vocationAllowance;
    }

    public double getGrossAllowance() {
        return baseAllowance + vocationAllowance;
    }

    public String toString() {
return "Commando{" + super.toString() + ", vocationAllowance=" + vocationAllowance + '}';
    }
}
